apps/cms-v2/src/modules/bulk-sync/bulk-sync-view.tsx

```diff
@@ -0,0 +1,173 @@
+import { Breadcrumbs } from "@saleor/apps-ui";
+import { ArrowRightIcon, Box, Button, Text } from "@saleor/macaw-ui/next";
+import { useEffect, useRef, useState } from "react";
+import { ChannelProviderConnectionConfig, ProvidersConfig } from "../configuration";
+import { AppHeader } from "../ui/app-header";
+import { AppSection } from "../ui/app-section";
+import { ButtonsBox } from "../ui/buttons-box";
+
+import { useBulkSyncProductsState } from "./use-bulk-sync-products-state";
+import { useFetchAllProducts } from "./use-fetch-all-products";
+import { VariantsSyncStatusList } from "./variants-sync-status-list";
+import { ProvidersResolver } from "../providers/providers-resolver";
+
+const FetchProductsStep = (props: { onButtonClick(): void }) => {
+  return (
+    <Box>
+      <Text variant="heading" as="h2" marginBottom={4}>
+        Saleor products fetch
+      </Text>
+      <Text as="p">Click the button to start fetching products from Saleor API</Text>
+      <Text as="p">After products are fetched, you will be able to upload them to the CMS</Text>
+      <ButtonsBox>
+        <Button onClick={props.onButtonClick}>Prefetch products</Button>
+      </ButtonsBox>
+    </Box>
+  );
+};
+
+const SaleorProductsFetchingStep = () => {
+  return (
+    <Box>
+      <Text variant="heading" as="h2" marginBottom={4}>
+        Saleor products fetch
+      </Text>
+      <Text as="p">Fetching...</Text>
+    </Box>
+  );
+};
+
+const SaleorProductsFetchedStep = (props: { productsNo: number; variantsNo: number }) => {
+  return (
+    <Box>
+      <Text variant="heading" as="h2" marginBottom={4}>
+        Saleor products fetch
+      </Text>
+      <Text as="p">
+        Fetched {props.productsNo} products and {props.variantsNo} variants
+      </Text>
+    </Box>
+  );
+};
+
+type State = "initial" | "fetching" | "fetched" | "uploading";
+
+export const BulkSyncView = ({
+  configuration,
+  connection,
+}: {
+  configuration: ProvidersConfig.AnyFullShape;
+  connection: ChannelProviderConnectionConfig.FullShape;
+}) => {
+  const processor = useRef(ProvidersResolver.createBulkSyncProcessor(configuration));
```

Comment:
To avoid creating new instances of this class

>useRef is a React Hook that lets you reference a value that’s not needed for rendering.


https://react.dev/reference/react/useRef